src/config/id.rs

```diff
@@ -0,0 +1,108 @@
+use serde::de::{self, Visitor};
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+use std::cmp::{Ord, Ordering, PartialOrd};
+use std::fmt;
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct ComponentId {
+    pub name: String,
```

Comment:
I know this is a bit repetitive with the struct name, but I might rename this field `id` to match how we switched to referring to component IDs in https://github.com/timberio/vector/pull/8417